core/oidc.go

```diff
@@ -492,6 +509,10 @@ func (o *OIDC) token(ctx context.Context, req *tokenRequest, handler func(req *T
 
 	tresp, err := handler(tr)
 	if err != nil {
+		var uaerr unauthorized = &unauthorizedErr{}
+		if errors.As(err, &uaerr); uaerr.Unauthorized() {
```

Comment:
```suggestion
		var uaerr unauthorized
		if errors.As(err, &uaerr); uaerr != nil && uaerr.Unauthorized() {
```

I think this might have some [strange behavior as written](https://goplay.space/#vx2c6ib6Gje) since `errors.As` will not touch `uaerr` if it's not an `unauthorized`

Is there a test we could tweak to have caught this case?

Reply:
Yeah that's right, I was being confused by some lint errors. Updated, and added a test case to catch this.